### PR TITLE
Implement replacing material with crystal. 

### PR DESCRIPTION
**Pull request by Unengine**
### Description

SSIA

### How to test

1. Check if replacing material with crystal is working properly.

### Required Reviewers
@sonohoshi @jaeho0103

### CODE REVIEW

**Review by boscohyun — APPROVED** [review]

File: nekoyume/Assets/_Scripts/UI/Combination/SubRecipeView.cs [hunk 1]
```diff
@@ -438,9 +456,9 @@ public void CombineCurrentRecipe()
             CombinationActionSubject.OnNext(_selectedRecipeInfo);
         }
 
-        private bool CheckMaterialAndSlot()
+        private bool CheckNCGAndSlot()
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Combination/SubRecipeView.cs:459]: It's a method to check if there are enough NCGs and slots, so why don't you add what you want to check to method name?
Like `CheckNCGAndSlotIsEnough`?

File: nekoyume/Assets/_Scripts/UI/Module/Item/ItemView.cs [hunk 2]
```diff
@@ -33,8 +33,6 @@ public class ItemView<TViewModel> : VanillaItemView
 #endif
         public TouchHandler touchHandler;
 
-        public Button itemButton;
-        public Image backgroundImage;
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Module/Item/ItemView.cs:37]: Is this intended?
If you intended it, these fields is not needs? Or controlled by other class?

Comment (Unengine) [reply nekoyume/Assets/_Scripts/UI/Module/Item/ItemView.cs:37]: These are not used.

Comment (sonohoshi) [reply nekoyume/Assets/_Scripts/UI/Module/Item/ItemView.cs:37]: 👌

File: nekoyume/Assets/_Scripts/UI/Widget/Craft.cs [hunk 3]
```diff
@@ -359,6 +378,34 @@ private void CombinationConsumableAction(SubRecipeView.RecipeInfo recipeInfo)
             StartCoroutine(CoCombineNPCAnimation(consumable, requiredBlockIndex, true));
         }
 
+        public static Dictionary<int, int> CheckMaterial(Dictionary<int, int> materials)
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Craft.cs:381]: As far as I understand, the role of this method seems to be to check the original materials and inventory, and then return the missing materials.
But this method signature alone seems difficult to determine what role it works.
I sugest to change the name of parameter and name of method.

```suggestion
        public static Dictionary<int, int> GetMissingMaterials(Dictionary<int, int> originMaterialsOfRecipe)
```

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/ReplaceMaterialPopup.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            var yes = L10nManager.Localize("UI_YES");
+            var no = L10nManager.Localize("UI_NO");
+            confirmCallback = cost <= States.Instance.CrystalBalance.MajorUnit ?
+                confirmCallback :
+                () => OnInsufficientCost(cost);
+
+            Show(content, yes, no, confirmCallback);
+        }
+
+        private void OnInsufficientCost(BigInteger cost)
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/ReplaceMaterialPopup.cs:79]: Can it change to static method?

**Review by sonohoshi — APPROVED** [review]

**Pull request merged**